libs/driver-adapters/executor/src/qc-test-worker/worker-query.ts

```diff
@@ -194,3 +210,98 @@ function getOperationResultInQeFormat(result: unknown) {
     return result
   }
 }
+
+function isPlainRawQuery(plan: QueryPlanNode): boolean {
+  switch (plan.type) {
+    case 'query':
+      return plan.args.type === 'rawSql'
+    case 'seq':
+      return plan.args.length === 1 && isPlainRawQuery(plan.args[0])
+    default:
+      return false
+  }
+}
+
+function serializeRawQueryResult(
```

Comment:
I think the low lever serialization should be separated into another file for clarity, maybe together with the Query Interpreter. It would play nicely if the constructor mentioned in my comment above is added.